src/lib/SHX.js

```diff
@@ -324,7 +324,14 @@ async function fetchSHLManifest(shlPayload, passcode) {
   });
 
   if (response.status === 401 && passcode) {
-	throw new PasscodeError("Passcode incorrect.");
+    const responseBody = await response.json();
```

Comment:
I'd wrap this section up in a try/catch so you can throw a generic PasscodeError if the error json isn't what you expect